From 766bc4e8288ae99e14842c882091e653d4af6517 Mon Sep 17 00:00:00 2001
From: cooper_huang <cooper_huang@aspeedtech.com>
Date: Tue, 26 May 2026 12:53:29 +0800
Subject: [PATCH] mtd: spi-nor: Add XMC XM25QH01D flash part

Add flash ID entry for XMC XM25QH01D SPI NOR flash.

Upstream-Status: Backport [SDK v11.03 migration]
Signed-off-by: cooper_huang <cooper_huang@aspeedtech.com>
Change-Id: Ic9f66437a52de40ec0571aa3b1eccbae8d09828c
Signed-off-by: rupeshs <rupeshs@ami.com>
---
 drivers/mtd/spi/spi-nor-ids.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mtd/spi/spi-nor-ids.c b/drivers/mtd/spi/spi-nor-ids.c
index 4b7b358eac..85b46c2101 100644
--- a/drivers/mtd/spi/spi-nor-ids.c
+++ b/drivers/mtd/spi/spi-nor-ids.c
@@ -586,6 +586,8 @@ const struct flash_info spi_nor_ids[] = {
 	       SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES) },
 	{ INFO("XM25QU512C", 0x204120, 0, 64 * 1024, 1024, SECT_4K | SPI_NOR_DUAL_READ |
 	       SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES) },
+	{ INFO("XM25QH01D", 0x204021, 0, 64 * 1024, 2048, SECT_4K | SPI_NOR_DUAL_READ |
+	       SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES) },
 #endif
 #ifdef CONFIG_SPI_FLASH_XTX
 	/* XTX Technology Limited */
-- 
2.43.0

